lib/types.go

```diff
@@ -41,6 +42,18 @@ type Options struct {
 	Thresholds map[string][]*Threshold `json:"thresholds"`
 }
 
+type SourceData struct {
+	SrcData  []byte
```

Comment:
`SourceData.SrcData` feels a little… redundant? `Data` seems to convey the same information with less. Same for `SrcType` and `Type`.